packages/next/next-server/lib/router/router.ts

```diff
@@ -824,6 +850,29 @@ export default class Router implements BaseRouter {
     return this.asPath !== asPath
   }
 
+  _resolveHref(parsedHref: UrlObject, pages: string[]) {
+    const { pathname } = parsedHref
+    const cleanPathname = denormalizePagePath(delBasePath(pathname!))
+
+    if (cleanPathname === '/404' || cleanPathname === '/_error') {
+      return parsedHref
+    }
+
+    // handle resolving href for dynamic routes
+    if (!pages.includes(cleanPathname!)) {
+      for (let page of pages) {
```

Comment:
Please use `forEach` instead so we can trim some bundle weight

Reply:
Just checked and unless we use arrow functions it will not result in a smaller output